common/views/pages/_app.js

```diff
@@ -13,29 +13,43 @@ import ErrorPage from '../../views/components/ErrorPage/ErrorPage';
 import TogglesContext from '../../views/components/TogglesContext/TogglesContext';
 import OpeningTimesContext from '../../views/components/OpeningTimesContext/OpeningTimesContext';
 import GlobalAlertContext from '../../views/components/GlobalAlertContext/GlobalAlertContext';
-
+import {trackEvent} from '../../utils/ga';
 const isServer = typeof window === 'undefined';
 const isClient = !isServer;
 
 let toggles;
 let openingTimes;
 let globalAlert;
-let engagement;
+let previousTimeOnPage = 0;
+let pageHiddenTime = 0;
+let pageVisibilityLastChanged = 0;
 
-function triggerEngagement() {
-  ReactGA.event({
+function trackVisibleTimeOnPage () {
+  const visibleTime = Math.round(window.performance.now() - previousTimeOnPage - pageHiddenTime);
+  trackEvent({
     category: 'Engagement',
-    action: 'Time on page >=',
-    label: '10 seconds'
+    action: 'Time page is visible',
```

Comment:
nitpick, but I think the rest of our new actions don't start with a capital letter?